storage/innobase/log/log0recv.cc

```diff
@@ -1830,6 +1830,16 @@ static byte *recv_parse_or_apply_log_rec_body(
             !fsp_is_system_or_temp_tablespace(space_id) &&
             /* For cloned db header page has the encryption information. */
             !recv_sys->is_cloned_db) {
+          /* If Encryption info is changed after the last checkpoint lSN, scan
+          the tablespace to know modified pages. This scan and copy of modified
+          pages would avoid the apply the redo logs due to changed in encryption
+          info. As there is problem of apply redo on page 0 because it reset
+          in-memory encryption and page can't decrypted */
```

Comment:
The point about scan and copying modified pages seems like an optimization. I think it is irrelevant to the current bug. I think this bug is because a full backup can have full encrypted tablespace and for the next incremental, if ALTER TABLESPACE ENCRYPTION='N' happens and if checkpoint doesn't move since fullbackup, it is possible that we apply page 0 encryption='n' change first. ie reset of encryption info first.

Then after this page 0 redo apply, we cannot read other encrypted pages from disk and cannot make them unencrypted.

Reply:
fixed